xdsl/transforms/experimental/ConvertStencilToLLMLIR.py

```diff
@@ -51,6 +52,26 @@ def match_and_rewrite(self, op: CastOp, rewriter: PatternRewriter, /):
         rewriter.replace_matched_op(memref.Cast.get(op.field, result_typ))
 
 
+class LoadOpToMemref(RewritePattern):
+
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: LoadOp, rewriter: PatternRewriter, /):
+        if op.lb is None:
+            warn("stencil.load should have a lb attribute when lowered.")
+            return
+        if not isinstance(op.field.owner, memref.Cast):
+            warn(
+                "stencil.cast should be lowered to memref.cast before the stencil.load lowering."
+            )
+            return
+        offsets: list[Attribute] = [
+            IntegerAttr(-i.value.data, i64) for i in op.lb.array.data
+        ]
+        op.field.owner.attributes["stencil_offset"] = ArrayAttr.from_list(
+            offsets)
+        rewriter.erase_matched_op()
```

Comment:
I don't know the dialects here well enough. but what is this actually rewriting? It seems to me it changes some Attributes (that should be frozen AFAIU) and then just erases the op. This can very well be what you want to do, but it looks weird to me :)

Reply:
It is a bit weird, and I switched to a modified version in a later commit, I'll see if it's a pain to pick in that PR!
I still mostly "just erase it" though. Reason is, the straightforward way to lower this operation is using memref.subview, basically declaring a custom accessor to an underlying memref. And we don't have those yet, and they are too costly to implement for the rush we want stencil for.

My workaround then is to mark the memref itself with offsets, to use them in later accesses, to code-generate the custom access myself with arith.

Reply:
Yeah, a TODO sounds good :+1: